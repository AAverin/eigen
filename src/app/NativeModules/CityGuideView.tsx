import React from "react"
import { requireNativeComponent } from "react-native"

const ARTCityGuideView = requireNativeComponent("ARTCityGuideView")

export const CityGuideView: React.FC = () => {
  return (
    <ARTCityGuideView // @ts-ignore
      style={{ flex: 1 }}
    />
  )
}
