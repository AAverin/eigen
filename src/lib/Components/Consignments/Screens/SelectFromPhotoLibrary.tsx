import * as React from "react"

import Button from "../../Buttons/FlatWhite"
import DoneButton from "../Components/BottomAlignedButton"

import Circle from "../Components/CircleImage"
import ConsignmentBG from "../Components/ConsignmentBG"

import ImageSelection, { ImageData } from "../Components/ImageSelection"
import { BodyText as P } from "../Typography"

import {
  CameraRoll,
  GetPhotosParamType,
  GetPhotosReturnType,
  NavigatorIOS,
  Route,
  ScrollView,
  View,
  ViewProperties,
} from "react-native"

interface Props extends ViewProperties {
  navigator: NavigatorIOS
  route: Route
}

interface State {
  cameraImages: ImageData[]
  lastCursor: string
  loadingMore: boolean
  noMorePhotos: boolean
}

const doneButtonStyles = {
  backgroundColor: "black",
  marginBottom: 20,
  paddingTop: 18,
  height: 56,
}

const uri1 = "https://d32dm0rphc51dk.cloudfront.net/Vq62IJMLbG1TuUjs-2fZCg/large.jpg"
const uri2 = "https://d32dm0rphc51dk.cloudfront.net/WAlGHmjlxTn3USMllNt4rA/large.jpg"

export default class Welcome extends React.Component<Props, State> {
  constructor() {
    super()
    this.state = {
      cameraImages: [],
      loadingMore: false,
      lastCursor: "",
      noMorePhotos: false,
    }

    // if (CameraRoll) {
    //   CameraRoll.getPhotos({ first: 5, assetType: "Photos" }).then(photos => {
    //     this.setState(() => {
    //       return {
    //         cameraImages: photos.edges.map(e => e.node),
    //       }
    //     })
    //   })
    // }
    this.tryPhotoLoad()
  }

  tryPhotoLoad() {
    if (!this.state.loadingMore) {
      this.setState({ loadingMore: true }, () => {
        this.loadPhotos()
      })
    }
  }

  loadPhotos() {
    const fetchParams: GetPhotosParamType = {
      first: 35,
      groupTypes: "SavedPhotos",
      assetType: "Photos",
      after: null,
    }

    if (this.state.lastCursor) {
      fetchParams.after = this.state.lastCursor
    }

    CameraRoll.getPhotos(fetchParams)
      .then(data => {
        this.appendAssets(data)
      })
      .catch(e => {
        console.log(e)
      })
  }

  appendAssets(data) {
    const assets = data.edges
    const nextState = {
      loadingMore: false,
      noMorePhotos: !data.page_info.has_next_page,
    }

    if (assets.length > 0) {
      console.log(assets)
      // nextState.lastCursor = data.page_info.end_cursor
      // nextState.assets = this.state.assets.concat(assets)
      // nextState.dataSource = this.state.dataSource.cloneWithRows(_.chunk(nextState.assets, 3))
    }

    this.setState(nextState)
  }

  endReached() {
    if (!this.state.noMorePhotos) {
      this.tryPhotoLoad()
    }
  }

  doneTapped = () => {
    this.props.navigator.pop()
  }

  onPressNewPhoto = () => {
    console.log("OK")
  }

  render() {
    return (
      <ConsignmentBG>
        <DoneButton onPress={this.doneTapped} bodyStyle={doneButtonStyles} buttonText="DONE">
          <ScrollView style={{ flex: 1 }}>
            <View style={{ paddingTop: 40, alignItems: "center" }}>
              <P>We suggest adding a few photos of the work including the front and back as well as the signature.</P>
              <ImageSelection data={this.state.cameraImages} onPressNewPhoto={this.onPressNewPhoto} />
            </View>
          </ScrollView>
        </DoneButton>
      </ConsignmentBG>
    )
  }
}
