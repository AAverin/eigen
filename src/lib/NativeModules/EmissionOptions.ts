import { isEqual } from "lodash"
import { useEffect, useState } from "react"
import { NativeEventEmitter, NativeModules } from "react-native"

const { Emission } = NativeModules

const emitter = new NativeEventEmitter(Emission as any)

let needsRefreshing = false

export const useEmissionOptions = () => {
  const [refreshedOptions, setRefreshedOptions] = useState(Emission.options)

  useEffect(() => {
    const listener = emitter.addListener("featuresDidChange", data => {
      console.log("wowowowow1")
      console.log({ arenableview1: data.AREnableViewingRooms })

      if (!isEqual(refreshedOptions, data)) {
        setRefreshedOptions(data)
      }

      return () => listener.remove()
    })
  })

  Emission.getFreshOptions((error, freshOptions) => {
    console.log("wowowowow2")
    console.log({ arenableview2: freshOptions.AREnableViewingRooms })
    if (!isEqual(refreshedOptions, freshOptions)) {
      setRefreshedOptions(freshOptions)
    }
  })

  console.log({ arenableview: refreshedOptions.AREnableViewingRooms })
  return refreshedOptions
}
