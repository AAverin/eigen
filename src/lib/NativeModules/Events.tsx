import { NativeModules } from "react-native"
const { AREventsModule } = NativeModules

function postEvent(info: any) {
  if (__DEV__) {
    console.log("[Event tracked]", JSON.stringify(info, null, 2))
  }
  AREventsModule.postEvent(info)
}

export default { postEvent }
