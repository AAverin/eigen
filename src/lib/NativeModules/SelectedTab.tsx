import { isEqual } from "lodash"
import React, { createContext, useCallback, useContext, useEffect, useState } from "react"
import { NativeEventEmitter, NativeModules } from "react-native"

export enum TabName {
  ARHomeTab = "ARHomeTab",
  ARSearchTab = "ARSearchTab",
  ARMessagingTab = "ARMessagingTab",
  ARLocalDiscoveryTab = "ARLocalDiscoveryTab",
  ARFavoritesTab = "ARFavoritesTab",
  ARSalesTab = "ARSalesTab",
  Unknown = "Unknown",
}

interface SelectedTab {
  name: string
}

const changes = new NativeEventEmitter(NativeModules.ARSelectedTab)

let initialTab: SelectedTab = {
  name: NativeModules.ARSelectedTab.name,
}

changes.addListener("selectedTabChanged", nextTab => {
  initialTab = {
    name: nextTab.name,
  }
})

export const SelectedTabContext = createContext(initialTab)

export const ProvideSelectedTab: React.FC = ({ children }) => {
  const [selectedTab, setSelectedTab] = useState(initialTab)

  const onChange = useCallback(
    (nextTab: { name: string }) => {
      if (!isEqual(selectedTab, nextTab)) {
        setSelectedTab(nextTab)
      }
    },
    [selectedTab]
  )

  useEffect(() => {
    changes.addListener("selectedTabChanged", onChange)
    return () => {
      changes.removeListener("selectedTabChanged", onChange)
    }
  }, [onChange])

  return <SelectedTabContext.Provider value={selectedTab}>{children}</SelectedTabContext.Provider>
}

export function useSelectedTab() {
  return useContext(SelectedTabContext)
}
