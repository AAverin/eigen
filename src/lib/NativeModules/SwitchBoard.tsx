import { findNodeHandle, NativeModules } from "react-native"
const { ARSwitchBoardModule } = NativeModules

function presentNavigationViewController(component: React.Component<any, any>, route: string) {
  let reactTag
  try {
    reactTag = findNodeHandle(component)
  } catch (err) {
    return
  }

  ARSwitchBoardModule.presentNavigationViewController(reactTag, route)
}

function presentModalViewController(component: React.Component<any, any>, route: string) {
  let reactTag
  try {
    reactTag = findNodeHandle(component)
  } catch (err) {
    return
  }

  ARSwitchBoardModule.presentModalViewController(reactTag, route)
}

function dismissModalViewController(component: React.Component<any, any>) {
  let reactTag
  try {
    reactTag = findNodeHandle(component)
  } catch (err) {
    return
  }

  ARSwitchBoardModule.dismissModalViewController(reactTag)
}

export default {
  presentNavigationViewController,
  presentModalViewController,
  dismissModalViewController,
}
