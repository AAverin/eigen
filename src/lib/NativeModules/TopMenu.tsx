import { NativeModules } from "react-native"

const { ARTopMenuModule } = NativeModules

export function getSelectedTabName() {
  return ARTopMenuModule.getSelectedTabName()
}
