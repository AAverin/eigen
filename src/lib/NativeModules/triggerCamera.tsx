import { findNodeHandle, NativeModules } from "react-native"
const { ARTakeCameraModule } = NativeModules

async function triggerCamera(component: React.Component<any, any>): Promise<any> {
  let reactTag
  try {
    reactTag = findNodeHandle(component)
  } catch (err) {
    console.error("could not find tag for ")
    return
  }
  return ARTakeCameraModule.triggerCameraModal(reactTag)
}

export default { triggerCamera }
