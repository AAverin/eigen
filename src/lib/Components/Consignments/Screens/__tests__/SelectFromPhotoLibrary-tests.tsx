import React from "react"
import "react-native"
import * as renderer from "react-test-renderer"

import SelectFromPhotoLibrary from "../SelectFromPhotoLibrary"

jest.mock("lib/NativeModules/triggerCamera", () => ({ triggerCamera: jest.fn() }))
import { triggerCamera } from "lib/NativeModules/triggerCamera"
const triggerMock = triggerCamera as jest.Mock<any>

const nav = {} as any
const route = {} as any

it("Sets up the right view hierarchy", () => {
  const tree = renderer
    .create(<SelectFromPhotoLibrary navigator={nav} route={route} setup={{}} updateWithPhotos={() => ""} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it("adds new photo to the list, and selects it", () => {
  const newPhoto = { image: { url: "https://image.com" } }
  triggerMock.mockImplementationOnce(() => Promise.resolve(newPhoto))

  const select = new SelectFromPhotoLibrary({
    navigator: nav,
    route,
    setup: { photos: [] },
    updateWithPhotos: () => "",
  })

  select.setState = jest.fn()

  expect.hasAssertions()
  return select.onPressNewPhoto().then(() => {
    expect(select.setState).toBeCalledWith({
      cameraImages: [{ image: { url: "https://image.com" } }],
      selection: expect.anything(),
    })
  })
})
