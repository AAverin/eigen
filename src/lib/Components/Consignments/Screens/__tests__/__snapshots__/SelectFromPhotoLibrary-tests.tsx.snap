// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sets up the right view hierarchy 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "black",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
          "backgroundColor": "black",
          "height": "100%",
          "width": "100%",
        },
      ]
    }
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "flexGrow": 1,
          }
        }
      >
        <RCTScrollView
          onScroll={[Function]}
          scrollEventThrottle={50}
          scrollsToTop={true}
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "paddingTop": 40,
                }
              }
            >
              <View
                px={2}
                style={
                  Array [
                    Object {
                      "paddingLeft": 20,
                      "paddingRight": 20,
                    },
                  ]
                }
              >
                <Text
                  color="#C2C2C2"
                  fontFamily="ReactNativeAGaramondPro-Regular"
                  fontSize="18px"
                  lineHeight="26px"
                  style={
                    Array [
                      Object {
                        "color": "#C2C2C2",
                        "fontFamily": "ReactNativeAGaramondPro-Regular",
                        "fontSize": 18,
                        "lineHeight": 26,
                      },
                      Object {},
                      Object {
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  We suggest adding a few photos of the work including the front and back as well as the signature.
                </Text>
              </View>
              <View
                mb={2}
                style={
                  Array [
                    Object {
                      "marginBottom": 20,
                    },
                  ]
                }
              />
              <RCTScrollView
                columnWrapperStyle={
                  Object {
                    "justifyContent": "space-between",
                    "paddingLeft": 20,
                    "paddingRight": 20,
                  }
                }
                data={
                  Array [
                    "take_photo",
                  ]
                }
                disableVirtualization={false}
                getItem={[Function]}
                getItemCount={[Function]}
                horizontal={false}
                initialNumToRender={10}
                keyExtractor={[Function]}
                maxToRenderPerBatch={10}
                numColumns={2}
                onContentSizeChange={[Function]}
                onEndReachedThreshold={2}
                onLayout={[Function]}
                onMomentumScrollEnd={[Function]}
                onScroll={[Function]}
                onScrollBeginDrag={[Function]}
                onScrollEndDrag={[Function]}
                removeClippedSubviews={false}
                renderItem={[Function]}
                scrollEventThrottle={50}
                stickyHeaderIndices={Array []}
                updateCellsBatchingPeriod={50}
                viewabilityConfigCallbackPairs={Array []}
                windowSize={21}
              >
                <View>
                  <View
                    onLayout={[Function]}
                    style={null}
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "flexDirection": "row",
                          },
                          Object {
                            "justifyContent": "space-between",
                            "paddingLeft": 20,
                            "paddingRight": 20,
                          },
                        ]
                      }
                    >
                      <View
                        accessible={true}
                        focusable={true}
                        isTVSelectable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "white",
                            "height": 345,
                            "justifyContent": "center",
                            "marginBottom": 20,
                            "opacity": 1,
                            "width": 345,
                          }
                        }
                      >
                        <Image
                          source={
                            Object {
                              "testUri": "../../../images/consignments/camera-black.png",
                            }
                          }
                        />
                      </View>
                    </View>
                  </View>
                </View>
              </RCTScrollView>
            </View>
          </View>
        </RCTScrollView>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#666666",
              "height": 1,
            },
          ]
        }
      />
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "black",
            "height": 44,
            "marginBottom": 20,
            "opacity": 1,
            "paddingTop": 18,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "fontFamily": "AvantGardeGothicITC",
                "fontSize": 14,
                "textAlign": "center",
              },
            ]
          }
        >
          Done
        </Text>
      </View>
    </View>
  </View>
</View>
`;
